src/core/server/http/http_service.ts

```diff
@@ -94,6 +105,31 @@ export class HttpService implements CoreService<HttpServiceSetup, HttpServiceSta
     };
   }
 
+  private async createServer(cfg: Partial<HttpConfig>) {
+    const { port } = cfg;
+
+    if (!port) {
+      throw new Error('port must be defined');
+    }
+
+    if (this.secondaryServers.has(port)) {
+      throw new Error(`port ${port} is already in use`);
+    }
+
+    for (const [key, val] of Object.entries(cfg)) {
+      httpConfig.schema.validateKey(key, val);
+    }
+
+    const baseConfig = await this.config$.pipe(first()).toPromise();
+    const finalConfig = { ...baseConfig, ...cfg };
+    const log = this.logger.get('http', `server:${port}`);
+
+    const httpServer = new HttpServer(log);
+    const httpSetup = await httpServer.setup(finalConfig);
```

Comment:
this PR highlights several problems with the current setup (can be addressed in a following PRs)
- end user has to provide config for both `setup` and `start`.
- http server is shipped with Kibana specific functionality, that shouldn't leak to end users. related https://github.com/elastic/kibana/pull/36690#discussion_r287815555 I can create a separate issue.

@elastic/kibana-platform any thoughts?